src/pyk/kcfg/kcfg.py

```diff
@@ -110,40 +111,67 @@ def pretty(self, kprint: KPrint, minimize: bool = True) -> Iterable[str]:
             ]
 
     @dataclass(frozen=True)
-    class Split(Successor):
+    class MultiEdge(Successor):
         source: KCFG.Node
-        targets: Tuple[Tuple[KCFG.Node, CSubst], ...]
+        targets: Tuple[KCFG.Node, ...]
+
+        @property
+        def target_ids(self) -> List[str]:
+            return sorted([t.id for t in self.targets])
+
+        def __lt__(self, other: Any) -> bool:
+            if not type(other) is type(self):
+                return NotImplemented
+            return (self.source, self.target_ids) < (other.source, other.target_ids)
+
+        @abstractmethod
+        def with_single_target(self, target: KCFG.Node) -> KCFG.MultiEdge:
+            ...
+
+    @dataclass(frozen=True)
+    class Split(MultiEdge):
+        splits: Dict[str, CSubst]
 
         def __init__(
             self,
             source: KCFG.Node,
             targets: Iterable[Tuple[KCFG.Node, CSubst]],
         ):
-            object.__setattr__(self, 'source', source)
-            object.__setattr__(self, 'targets', tuple(targets))
-
-        def __lt__(self, other: Any) -> bool:
-            if not isinstance(other, KCFG.Split):
-                return NotImplemented
-            return (self.source, self.target_ids) < (other.source, other.target_ids)
+            targets = list(targets)
+            super().__init__(source, tuple(n for n, _ in targets))
+            object.__setattr__(self, 'splits', {n.id: s for n, s in targets})
 
         def to_dict(self) -> Dict[str, Any]:
             return {
                 'source': self.source.id,
-                'targets': {target.id: csubst.to_dict() for target, csubst in self.targets},
+                'targets': {target.id: self.splits[target.id].to_dict() for target in self.targets},
             }
 
-        @property
-        def target_ids(self) -> List[str]:
-            return sorted(t.id for t, _ in self.targets)
+        def pretty(self, kprint: KPrint) -> List[str]:
+            return [f'Split node: {len(self.targets)}']
+
+        def with_single_target(self, target: KCFG.Node) -> KCFG.Split:
+            return KCFG.Split(self.source, [(target, self.splits[target.id])])
+
+    @dataclass(frozen=True)
+    class NDBranch(MultiEdge):
+        def to_dict(self) -> Dict[str, Any]:
```

Comment:
```suggestion
    @dataclass(frozen=True)
    class NDBranch(MultiEdge):
        source: KCFG.Node
        targets: Tuple[KCFG.Node, ...]

        def to_dict(self) -> Dict[str, Any]:
```

Actually not sure if this is needed, but I've seen redeclaration of the fields on other dataclasses. Maybe @tothtamas28 can comment?

Reply:
`MultiEdge` is a dataclass itself, so it is not needed. But it is also abstract, so I think it shouldn't be a dataclass.